### ISSUE TITLE
create_neurites() not knowing axon

### ISSUE THREAD

**Issue by SBottani** [SENeC-Initiative/DeNSE#24]
Hi again,

Using the new neurite parameter dictionary

    neurite_params = {"axon": axon_params, "dendrite": dendrite_params}

 when attempting to use _create_neurites()_ on neurons to create an axon and neurites, the "axon" entry in the neurite parameters is not recognized:

```python
# Here neurons created
gids = ds.create_neurons(n=num_neurons, params=neuron_params,  num_neurites=0)

# Now for each neuron, creation of neurites
for neuron in gids:
    neuron.create_neurites(num_neurites=2, params=neurite_params)

```
Here are the definition of the parameters
```python
neuron_params = {
    "axon_diameter": 4. * um,
    "dendrite_diameter": 2.*um,
    "soma_radius": soma_radius * um,
}

axon_params = {
    "growth_cone_model": gc_model,
    "use_uniform_branching": use_uniform_branching,
    "use_van_pelt": use_vp ,
    "sensing_angle": 45.*deg,
    "speed_growth_cone": 0.025 * um / minute, #0.15
    "filopodia_wall_affinity": 100.,
    "filopodia_finger_length": 7. * um,
    "filopodia_min_number": 30,
    "persistence_length": 300. * um,
    "taper_rate": 1./4000., 
    'B': 3. * cpm,
    'T': 1000. * minute,
    'E': 1.,   
}

dendrite_params = {
    "use_van_pelt": use_vp,
    "growth_cone_model": gc_model,
    "speed_growth_cone": 0.01 * um / minute,
    "filopodia_wall_affinity": 10. ,
    "persistence_length" : 200. * um,
    "taper_rate": 3./250.,
    "B": 6. * cpm,
    "T": 1000. * minute,
    'E': 1.,
}

neurite_params = {"axon": axon_params, "dendrite": dendrite_params}
```

**Comment by SBottani** [comment]
Axons and dendrtie must be **separatedly** created.

Building on the issue above.

The following code works, with axon created separatedly from dendrites

 ```
   for neuron in gids:
        # neuron.create_neurites(num_neurites=3,
        #                        params=neurite_params,
        #                        neurite_types=["axon", "dendrite", "dendrite"])
        # Add axon
        neuron.create_neurites(num_neurites=1,
                               params=axon_params,
                               neurite_types="axon")
        # Add dendrites
        neuron.create_neurites(num_neurites=2,
                               params=dendrite_params,
                               neurite_types=["dendrite","dendrite"])
```

Attempted also the following code but did not work
```
    for neuron in gids:
        neuron.create_neurites(num_neurites=3,
                               params=neurite_params,
                               neurite_types=["axon", "dendrite", "dendrite"])
```
whereby
`neurite_params = {"axon": axon_params, "dendrite": dendrite_params}`

**Comment by Silmathoron** [comment]
Hi @SBottani 
it indeed seems that ``create_neurites`` currently still follows the old structure and does not expect named neurites in the parameters, I will fix this ASAP.

In the meantime, the code for creating neurons and neurites in one go, with separate parameters is
```python
neuron = ds.create_neurons(params=neuron_params, neurite_params=neurite_params,
                           num_neurites=2)
```
This should be explained in the [tutorial](https://dense.readthedocs.io/en/latest/user/tutorial.html), please let me know if you think it is unclear and how we should improve it.

**Comment by SBottani** [comment]
Hi again Tanguy,

I agree with the `ds.create_neurons() ` method and it is well explained in the tutorial.
Yet `ds.create_neurite()` seems necessary if you want to  control also the initial neurite angles. Not possible with `ds.create_neurons() ` alon isn't it ?

Samuel

**Comment by SBottani** [comment]
Oh, I think I'm figuring out something.... !

You can set initial neurite angles in `ds.create_neurons()` ! With parameter `neurite_angles`

You can set it with d`s.create_neurite()`, with parameter `angles`. 

Maybe the same naming would help. Sorry for the confusion.

**Comment by Silmathoron** [comment]
Yes, that is how it works.
I don't think using the same naming is a good idea here:
* ``neurite_angles`` is necessary in ``create_neurons`` because we are creating neurons so we need to explicit what are the angles
* in ``create_neurites``, we are creating neurites, so the angles cannot refer to anything but the neurites, hence the simpler ``angles``

Maybe we could add this explanation in the tutorial, though.

**Comment by Silmathoron** [comment]
Solved by #29
